.vib/memcached/goss/memcached.yaml

```diff
@@ -0,0 +1,5 @@
+command:
+  # Perform a simple spin-up of memcached as a deamon
+  check-memcached-daemon:
+    exec: memcached -u {{ .Vars.memcached.user }} -p {{ .Vars.memcached.port }} -d
+    exit-status: 0
```

Comment:
This test is prone to false positives. When a process is launched as a daemon, the exit status does not really reflect if the operation succeeded (it shows whether the daemon process was created or not instead):

```bash
$ memcached -u 1001 -p 11211
failed to listen on TCP port 11211: Address already in use
$ echo $?
71
$ memcached -u 1001 -p 11211 -d
$ echo $?
0
```

To ensure this works as expected, I would use `timeout` and avoid using the _daemon_ feature:

```bash
$ timeout --preserve-status 2s memcached -u 1001 -p 11211
failed to listen on TCP port 11211: Address already in use
$ echo $?
71
```

Reply:
Nice catch, thanks!